### PR TITLE
Add support for Sve.UnzipEven/Odd & Sve.ZipHighLow

### PR DESCRIPTION
**Pull request by SwapnilGaikwad**
Add SVE API for unpredicated versions of UnzipEven, UnzipOdd, ZipHigh & ZipLow methods.

### CODE REVIEW

**Comment by SwapnilGaikwad** [comment]
On a V1 system with SVE vector length 256 bits, tests for this patch would fail. We are currently limited to vector length of 128 causing incorrect results on systems with higher vector lengths. We need #101295 to limit vector length to 128 bits.

**Comment by JulieLeeMSFT** [comment]
Contributes to https://github.com/dotnet/runtime/issues/99957

**Comment by JulieLeeMSFT** [comment]
Thanks @SwapnilGaikwad for the PR for bitmanipulate. When you merge the PR, could you check off the boxes from https://github.com/dotnet/runtime/issues/99957?

**Comment by kunalspathak** [comment]
> We need #101295 to limit vector length to 128 bits.

Is your plan to *always* limit the vector length to 128 bits? If yes, how can we test the functionality on 256 bits and higher?

File: src/tests/JIT/HardwareIntrinsics/Arm/Shared/_BinaryOpSveTestTemplate.template [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using Xunit;
+
+namespace JIT.HardwareIntrinsics.Arm
+{
```
Comment (kunalspathak) [inline src/tests/JIT/HardwareIntrinsics/Arm/Shared/_BinaryOpSveTestTemplate.template:20]: We might add some redundant templates for API testing, so I suggest for easy discoverability to use naming convention of `_Sve*.template`.

**Review by kunalspathak — COMMENTED** [review]
the changes looks good otherwise, except the comment about testing.

**Comment by SwapnilGaikwad** [comment]
> Thanks @SwapnilGaikwad for the PR for bitmanipulate. When you merge the PR, could you check off the boxes from #99957?

I'll post a comment on the issue as I cannot edit the text of the issue 👍

File: src/tests/JIT/HardwareIntrinsics/Arm/Shared/_BinaryOpSveTestTemplate.template (hunk 1 above)
Comment (SwapnilGaikwad) [reply src/tests/JIT/HardwareIntrinsics/Arm/Shared/_BinaryOpSveTestTemplate.template]: Also updated the generated template to `SveVecPairBinOpTest.template` to match the existing `SveLoadMaskedUnOpTest.template`.

**Review by kunalspathak — APPROVED** [review]
LGTM

**Comment by kunalspathak** [comment]
The only failure that BuildAnalysis didn't tag as "known issue" is actually https://github.com/dotnet/runtime/issues/91757 and not sure why it didn't get tagged. cc: @JulieLeeMSFT

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs [hunk 2]
```diff
@@ -861,5 +861,287 @@ public new abstract class Arm64 : AdvSimd.Arm64
...
 
+        ///  UnzipEven : Concatenate even elements from two inputs
+
+        /// <summary>
+        /// svuint8_t svuzp1[_u8](svuint8_t op1, svuint8_t op2)
+        ///   UZP1 Zresult.B, Zop1.B, Zop2.B
+        /// svbool_t svuzp1_b8(svbool_t op1, svbool_t op2)
+        ///   UZP1 Presult.B, Pop1.B, Pop2.B
+        /// </summary>
+        public static unsafe Vector<byte> UnzipEven(Vector<byte> left, Vector<byte> right) => UnzipEven(left, right);
```
Comment (kunalspathak) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs:872]: I just realized that zip and unzip instructions operate on both vector and predicate registers. How do we know which one to invoke? Currently, we will always invoke vector variants? @a74nh ?

If we always going to support just the vector version, then please remove the comment for predicate registers.

Comment (a74nh) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs:872]: Right now it'll just be the vector variant. If we're using masks then there will be a lot of conversions and the vector version will be used.

Raised a ticket: https://github.com/dotnet/runtime/issues/101598

In the meantime, is it ok to leave the comment as is?

Comment (kunalspathak) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs:872]: Thanks for opening an issue and yes, ok to keep it as of now.

**Pull request merged**